src/theory/tools/wp/bir_htScript.sml

```diff
@@ -0,0 +1,614 @@
+open HolKernel Parse boolLib bossLib;
+
+(* From /core: *)
+open bir_programTheory bir_typing_progTheory bir_envTheory
+     bir_auxiliaryTheory bir_valuesTheory bir_expTheory
+     bir_exp_immTheory bir_typing_expTheory bir_immTheory;
+
+(* From /theories: *)
+open bir_program_blocksTheory bir_program_terminationTheory
+     bir_program_valid_stateTheory bir_exp_substitutionsTheory
+     bir_bool_expTheory bir_program_env_orderTheory
+     bir_program_multistep_propsTheory bir_exp_equivTheory
+     bir_subprogramTheory;
+
+open bir_wpTheory;
+
+open HolBACoreSimps;
+
+val _ = new_theory "bir_ht";
+
+val bir_stmtb_is_not_assume_def = Define `
+  (bir_stmtb_is_not_assume (BStmt_Declare v) = T) /\
+  (bir_stmtb_is_not_assume (BStmt_Assign v ex) = T) /\
+  (bir_stmtb_is_not_assume (BStmt_Assert ex) = T) /\
+  (bir_stmtb_is_not_assume (BStmt_Assume ex) = F) /\
+  (bir_stmtb_is_not_assume (BStmt_Observe ec el obf) = T)
+`;
+
+val bir_stmtb_not_assume_never_assviol =
```

Comment:
This should be named `bir_stmtB_not_assume_never_assviol`. Also, as a French, I would prefer `assumviol` :sweat_smile:

Reply:
hahaha, oh yes, i think we can avoid this wrong interpretation by renaming

Reply:
Okay, I will replace all instances of "assviol" with "assumviol", sparing posteriority from having to read ambiguous abbreviations.

Reply:
Plus the ambiguity with assert :)